std/algorithm/mutation.d

```diff
@@ -2068,6 +2148,45 @@ if (isBidirectionalRange!Range
     return result;
 }
 
+///ditto
+// SwapStrategy s is ignored here, it's just always stable,
+// unstable doesn't really work with variable length encoding
+Range remove(alias pred, SwapStrategy s = SwapStrategy.stable, Range)
+(Range range)
+if (isNarrowString!Range
+    && isMutable!(typeof(range[0])))
+{
+    import std.utf : decode;
+    import std.functional : unaryFun;
+
+    alias pred_ = unaryFun!pred;
+
+    size_t charIdx = 0;
+    size_t charShift = 0;
+    while (charIdx < range.length)
+    {
+        size_t start = charIdx;
+        if (pred_(decode(range, charIdx)))
+        {
+            charShift += charIdx - start;
+            break;
+        }
+    }
+    while (charIdx < range.length)
+    {
+        size_t start = charIdx;
+        auto doRemove = pred_(decode(range, charIdx));
+        auto encodedLen = charIdx - start;
+        if (doRemove)
+            charShift += encodedLen;
+        else
+            foreach (i; start .. charIdx)
+                range[i - charShift] = range[i];
+    }
+
+    return range[0 .. $ - charShift];
```

Comment:
this looks very similar to the range-based `remove`. Why can't this be combined?

Reply:
I guess there might be a way to merge them, but the major difference is that for auto-decoded strings you can't just `move` elements around.